src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -893,5 +893,72 @@ protected static ImmutableArray<CodeAction> GetNestedActions(ImmutableArray<Code
 
             await TestActionCountAsync(input, outputs.Length, parameters);
         }
+
+        protected static void GetDocumentAndSelectSpanOrAnnotatedSpan(
+            TestWorkspace workspace,
+            out Document document,
+            out TextSpan span,
+            out string annotation)
+        {
+            annotation = null;
+            if (!TryGetDocumentAndSelectSpan(workspace, out document, out span))
+            {
+                document = GetDocumentAndAnnotatedSpan(workspace, out annotation, out span);
+            }
+        }
+
+        private static bool TryGetDocumentAndSelectSpan(TestWorkspace workspace, out Document document, out TextSpan span)
```

Comment:
Existing code moved down to the shared base type between refactoring and code fix tests. https://github.com/dotnet/roslyn/blob/c9d9e122cfff99ce9508a54897254d407b148f48/src/EditorFeatures/DiagnosticsTestUtilities/Diagnostics/AbstractUserDiagnosticTest.cs#L118-L141